### DISCUSSION TITLE
alternate stress and strain control in mtest

### DISCUSSION THREAD

**Discussion by GhiathMonnet** [thelfer/tfel#623] [General]
Dear All, 
I am looking for commands in mtest that simulate a first step in which a given strain rate (up to time t1) is imposed followed by a complete and fast unloading (stress becomes zero after t1). I need to predict the behavior after the fast unloading during a period, say t2. I know how to apply the first step. Something like :
@ImposedStrain<evolution> 'EXX' {0. : 0., t1 : 0.1};
@Times {0, t1 in 100};
But how can I add the unloading step ? The aim is to simulate the recovery rate during t2 - t1 of course. I keep getting an error message when using the @ImposedStress command.
Many thanks.
Ghiath

**Comment by thelfer** [comment]
Hi @GhiathMonnet,

Are you searching for a way to activate/desactivate a loading ?

Something like:

~~~c++
@ImposedDeformationGradient 'FRR' {0:1,1:'Fth',2:'Fth',3:'Fth-2*1.e-4/h0'}{
  desactivating_events : {'Stop'}
};

@Times {0,1,2, 3 in 200,3.01, 4 in 50};
@Event 'Stop' 3;
~~~

**Reply by thelfer** [reply]
Note that you also have the option `activating_events` to activate a loading.

**Comment by GhiathMonnet** [comment]
Thank you Thomas.
I don't know exactly what you mean by "desactivate event". But may be this is what I am looking for.
What I try to simulate is a mechanical test of polymer in which, the stress is measured during loading at constant strain rate up to a given strain (say 0.05). Then the sample is abruptly unloaded, but we keep monitoring the evolution of the deformation in the unloaded sample (recovery) in order to measure the residual deformation (after complete viscous relaxation).
What would the mtest command that I must use following the @ImposedStrain command ?

**Reply by thelfer** [reply]
This seems exactly what events are meant for.  In your case, you can do something like:

~~~c++
@ImposedStrain 'EXX' {0:1,1:1e-2'}{
  desactivating_events : {'Stop'}
};

@Times {0,1, 100 in 200};
@Event 'Stop' 1;
~~~

You will then impose the strain between times 0 and 1 and then release the constraint on `EXX` at time 1. The point falls back to satisfy the default condition `SXX=0`.

**Reply by thelfer** [reply]
Note that this is not very well tested. Do not hesitate to make feed-backs.

**Comment by GhiathMonnet** [comment]
Great. Many thanks Thomas. I will try that.
Cheers.

**Comment by GhiathMonnet** [comment]
It works very well. Many thanks Thomas.

**Reply by thelfer** [reply]
Great ! I'll have to document that properly !
